query.go

```diff
@@ -57,15 +57,9 @@ type queryFunc func(context.Context, peer.ID) (*dhtQueryResult, error)
 
 // Run runs the query at hand. pass in a list of peers to use first.
 func (q *dhtQuery) Run(ctx context.Context, peers []peer.ID) (*dhtQueryResult, error) {
-	select {
-	case <-ctx.Done():
+	if ctx.Err() != nil {
 		return nil, ctx.Err()
-	default:
 	}
-
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
 	runner := newQueryRunner(q)
 	return runner.Run(ctx, peers)
```

Comment:
So, I did a quick sanity check and `dhtQueryRunner.Run` is broken. _It_ should be creating a cancelable context and canceling it but it's not. See: https://github.com/libp2p/go-libp2p-kad-dht/blob/931ed2291c9daabc1d62ed9e305d957fbc3d6410/query.go#L122-L123

Reply:
@Stebalien thanks! I'll look into it more and fix that more directly.